File: ruby/raindrops/raindrops.rb
class Raindrops
  VERSION = 1

  SOUNDS = {
    3 => "Pling",
    5 => "Plang",
    7 => "Plong"
  }

  def self.convert(number)
    primes = SOUNDS.keys.select { |n| number % n == 0 }.map { |pn| SOUNDS[pn] }.join
    primes.empty? ? number.to_s : primes
  end


end

File: lua/raindrops/raindrops.lua
local raindrops = {}

local function raindrops(num)
  local result = ''
  if num % 3 == 0 then result = result .. 'Pling' end
  if num % 5 == 0 then result = result .. 'Plang' end
  if num % 7 == 0 then result = result .. 'Plong' end
  if result == '' then 
    return result .. num 
  else
    return result
  end
end

return raindrops
